api/v3/Case.php

```diff
@@ -101,31 +131,46 @@ function civicrm_api3_case_create($params) {
     CRM_Case_BAO_CaseContact::create($contactParams);
   }
 
+  if (!isset($params['id'])) {
+    // Creating a new case
+    _civicrm_api3_case_create_xmlProcessor($params, $caseBAO);
+  }
+
+  // return case
+  $values = array();
+  _civicrm_api3_object_to_array($caseBAO, $values[$caseBAO->id]);
+
+  // Add custom data
+  if (isset($params['custom'])) {
+    _civicrm_api3_custom_data_get($values[$caseBAO->id],TRUE,'case',$caseBAO->id);
+  }
+
+  return civicrm_api3_create_success($values, $params, 'Case', 'create', $caseBAO);
+}
+
+function _civicrm_api3_case_create_xmlProcessor($params, $caseBAO) {
+  // Format params for xmlProcessor
+  if (isset($caseBAO->id)) { $params['id'] = $caseBAO->id; }
+
   // Initialize XML processor with $params
   $xmlProcessor = new CRM_Case_XMLProcessor_Process();
   $xmlProcessorParams = array(
-    'clientID' => $params['contact_id'],
-    'creatorID' => $params['creator_id'],
+    'clientID' => CRM_Utils_Array::value('contact_id',$params),
+    'creatorID' => CRM_Utils_Array::value('creator_id',$params),
     'standardTimeline' => 1,
     'activityTypeName' => 'Open Case',
-    'caseID' => $caseBAO->id,
-    'subject' => $params['subject'],
-    'location' => CRM_Utils_Array::value('location', $params),
-    'activity_date_time' => $params['start_date'],
-    'duration' => CRM_Utils_Array::value('duration', $params),
-    'medium_id' => CRM_Utils_Array::value('medium_id', $params),
-    'details' => CRM_Utils_Array::value('details', $params),
+    'caseID' => CRM_Utils_Array::value('id',$params),
+    'subject' => CRM_Utils_Array::value('subject',$params),
+    'location' => CRM_Utils_Array::value('location',$params),
+    'activity_date_time' => CRM_Utils_Array::value('start_date',$params),
```

Comment:
There should be a space between the params of every `CRM_Utils_Array::value()` call here